commit 81cb02e2c22bbc036cdfaa2d2c4176f6bd60d3cf
Author: Jia Tan <jiat0218@gmail.com>
Date:   Thu Jan 12 23:43:06 2023 +0800

    CI: Disable shared and nls from various jobs in autotool runners.
    
    Disabling shared library generation and linking should help speed up the
    runners. The shared library is still being tested in the 32 bit build
    and the full feature.
    
    Disabling nls is to check for any unexpected warnings or errors.

diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index 8865a966..bbf24d81 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1,152 +1,152 @@
 #############################################################################
 #
 # Author: Jia Tan
 #
 # This file has been put into the public domain.
 # You can do whatever you want with this file.
 #
 #############################################################################
 
 name: CI
 
 on:
   # Triggers the workflow on push or pull request events but only for the master branch
   push:
     branches: [ master ]
   pull_request:
     branches: [ master ]
 
   # Allows running workflow manually
   workflow_dispatch:
 
 jobs:
   # When CMake can support disabling check types and
   # encoders/decoders/threads, it will be combined to one Linux job
   # and another matrix list.
   Linux-Autotools:
     # Just run on latest ubuntu
     runs-on: ubuntu-latest
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
       - name: Install Dependencies
         run: sudo apt-get install -y autoconf automake build-essential po4a autopoint gcc-multilib
 
       # -b specifies the build system to use.
       # -p specifies the phase (build or test) to help narrow down an error
       #    if one occurs.
       - name: Build with full features
         run: ./build-aux/ci_build.sh -b autotools -p build
       - name: Test with full features
         run: ./build-aux/ci_build.sh -b autotools -p test
 
       - name: Build 32-bit
         run: ./build-aux/ci_build.sh -b autotools -p build -f "-m32"
       - name: Test 32-bit
         run: ./build-aux/ci_build.sh -b autotools -p test -f "-m32"
 
       - name: Build without encoders
-        run: ./build-aux/ci_build.sh -b autotools -d encoders -p build
+        run: ./build-aux/ci_build.sh -b autotools -d encoders,shared -p build
       - name: Test without encoders
-        run: ./build-aux/ci_build.sh -b autotools -d encoders -p test
+        run: ./build-aux/ci_build.sh -b autotools -d encoders,shared -p test
 
       - name: Build without decoders
-        run: ./build-aux/ci_build.sh -b autotools -d decoders -p build
+        run: ./build-aux/ci_build.sh -b autotools -d decoders,shared -p build
       - name: Test without decoders
-        run: ./build-aux/ci_build.sh -b autotools -d decoders -p test
+        run: ./build-aux/ci_build.sh -b autotools -d decoders,shared -p test
 
       - name: Build without threads
-        run: ./build-aux/ci_build.sh -b autotools -d threads -p build
+        run: ./build-aux/ci_build.sh -b autotools -d threads,shared -p build
       - name: Test without threads
-        run: ./build-aux/ci_build.sh -b autotools -d threads -p test
+        run: ./build-aux/ci_build.sh -b autotools -d threads,shared -p test
 
       - name: Build without BCJ filters
-        run: ./build-aux/ci_build.sh -b autotools -d bcj -p build
+        run: ./build-aux/ci_build.sh -b autotools -d bcj,shared,nls -p build
       - name: Test without BCJ filters
-        run: ./build-aux/ci_build.sh -b autotools -d bcj -p test
+        run: ./build-aux/ci_build.sh -b autotools -d bcj,shared,nls -p test
 
       - name: Build without Delta filters
-        run: ./build-aux/ci_build.sh -b autotools -d delta -p build
+        run: ./build-aux/ci_build.sh -b autotools -d delta,shared,nls -p build
       - name: Test without Delta filters
-        run: ./build-aux/ci_build.sh -b autotools -d delta -p test
+        run: ./build-aux/ci_build.sh -b autotools -d delta,shared,nls -p test
 
       - name: Build without sha256 check
-        run: ./build-aux/ci_build.sh -b autotools -c crc32,crc64 -p build
+        run: ./build-aux/ci_build.sh -b autotools -c crc32,crc64 -d shared,nls -p build
       - name: Test without sha256 check
-        run: ./build-aux/ci_build.sh -b autotools -c crc32,crc64 -p test
+        run: ./build-aux/ci_build.sh -b autotools -c crc32,crc64 -d shared,nls -p test
 
       - name: Build without crc64 check
-        run: ./build-aux/ci_build.sh -b autotools -c crc32,sha256 -p build
+        run: ./build-aux/ci_build.sh -b autotools -c crc32,sha256 -d shared,nls -p build
       - name: Test without crc64 check
-        run: ./build-aux/ci_build.sh -b autotools -c crc32,sha256 -p test
+        run: ./build-aux/ci_build.sh -b autotools -c crc32,sha256 -d shared,nls -p test
 
   Linux-CMake:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - name: Install Dependencies
         run: sudo apt-get install -y build-essential cmake
       - name: Build
         run: ./build-aux/ci_build.sh -b cmake -p build
       - name: Test
         run: ./build-aux/ci_build.sh -b cmake -p test
 
   MacOS-Autotools:
     runs-on: macos-latest
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
       - name: Install Dependencies
         run: brew install autoconf automake libtool po4a
 
       - name: Build with full features
         run: ./build-aux/ci_build.sh -b autotools -p build
       - name: Test with full features
         run: ./build-aux/ci_build.sh -b autotools -p test
 
       - name: Build without encoders
-        run: ./build-aux/ci_build.sh -b autotools -d encoders -p build
+        run: ./build-aux/ci_build.sh -b autotools -d encoders,shared -p build
       - name: Test without encoders
-        run: ./build-aux/ci_build.sh -b autotools -d encoders -p test
+        run: ./build-aux/ci_build.sh -b autotools -d encoders,shared -p test
 
       - name: Build without decoders
-        run: ./build-aux/ci_build.sh -b autotools -d decoders -p build
+        run: ./build-aux/ci_build.sh -b autotools -d decoders,shared -p build
       - name: Test without decoders
-        run: ./build-aux/ci_build.sh -b autotools -d decoders -p test
+        run: ./build-aux/ci_build.sh -b autotools -d decoders,shared -p test
 
       - name: Build without threads
-        run: ./build-aux/ci_build.sh -b autotools -d threads -p build
+        run: ./build-aux/ci_build.sh -b autotools -d threads,shared -p build
       - name: Test without threads
-        run: ./build-aux/ci_build.sh -b autotools -d threads -p test
+        run: ./build-aux/ci_build.sh -b autotools -d threads,shared -p test
 
       - name: Build without BCJ filters
-        run: ./build-aux/ci_build.sh -b autotools -d bcj -p build
+        run: ./build-aux/ci_build.sh -b autotools -d bcj,shared,nls -p build
       - name: Test without BCJ filters
-        run: ./build-aux/ci_build.sh -b autotools -d bcj -p test
+        run: ./build-aux/ci_build.sh -b autotools -d bcj,shared,nls -p test
 
       - name: Build without Delta filters
-        run: ./build-aux/ci_build.sh -b autotools -d delta -p build
+        run: ./build-aux/ci_build.sh -b autotools -d delta,shared,nls -p build
       - name: Test without Delta filters
-        run: ./build-aux/ci_build.sh -b autotools -d delta -p test
+        run: ./build-aux/ci_build.sh -b autotools -d delta,shared,nls -p test
 
       - name: Build without sha256 check
-        run: ./build-aux/ci_build.sh -b autotools -c crc32,crc64 -p build
+        run: ./build-aux/ci_build.sh -b autotools -c crc32,crc64 -d shared,nls -p build
       - name: Test without sha256 check
-        run: ./build-aux/ci_build.sh -b autotools -c crc32,crc64 -p test
+        run: ./build-aux/ci_build.sh -b autotools -c crc32,crc64 -d shared,nls -p test
 
       - name: Build without crc64 check
-        run: ./build-aux/ci_build.sh -b autotools -c crc32,sha256 -p build
+        run: ./build-aux/ci_build.sh -b autotools -c crc32,sha256 -d shared,nls -p build
       - name: Test without crc64 check
-        run: ./build-aux/ci_build.sh -b autotools -c crc32,sha256 -p test
+        run: ./build-aux/ci_build.sh -b autotools -c crc32,sha256 -d shared,nls -p test
 
   MacOS-CMake:
     runs-on: macos-latest
     steps:
       - uses: actions/checkout@v2
       - name: Install Dependencies
         run: brew install cmake
       - name: Build
         run: ./build-aux/ci_build.sh -b cmake -p build
       - name: Test
         run: ./build-aux/ci_build.sh -b cmake -p test
     
\ No newline at end of file